back/app/models/static_page.rb

```diff
@@ -20,30 +37,69 @@
 class StaticPage < ApplicationRecord
   CODES = %w[about terms-and-conditions privacy-policy faq proposals custom].freeze
 
+  has_many :pins, as: :page, inverse_of: :page, dependent: :destroy
+  has_many :pinned_admin_publications, through: :pins, source: :admin_publication
   has_one :nav_bar_item, dependent: :destroy
   has_many :static_page_files, -> { order(:ordering) }, dependent: :destroy
   has_many :text_images, as: :imageable, dependent: :destroy
   accepts_nested_attributes_for :nav_bar_item
   accepts_nested_attributes_for :text_images
 
+  before_validation :set_code, on: :create
+  before_validation :generate_slug, on: :create
+
+  before_validation :strip_title
+  before_validation :sanitize_top_info_section_multiloc
+  before_validation :sanitize_bottom_info_section_multiloc
+
+  before_destroy :confirm_is_custom, prepend: true do
+    throw(:abort) if errors.present?
+  end
```

Comment:
[StackOverflow](https://stackoverflow.com/questions/123078/how-do-i-validate-on-destroy-in-rails#:~:text=State%20of-,affairs,-as%20of%20Rails)

This works, but I'm not sure it's our convention, as I haven't yet found another example in our codebase.
Maybe should be in a `before destroy` method in `side_fx_static_page_service.rb`, but again, I've not yet found a similar example in our codebase.

Returns a `Status: 500, "Internal Server Error"` when deletion is prevented because the `:code` is not `custom`, as this seems to be the desired [behaviour in the controller](https://github.com/CitizenLabDotCo/citizenlab/blob/e938b2be75464d1919edf4a7ec65f108feab663e/back/app/controllers/web_api/v1/static_pages_controller.rb#L58).

Reply:
Doesn't returning false work? https://stackoverflow.com/a/39280702
It would not throw an error, and so there will be no `false.destroyed?` (maybe?), and so no 500 code. I think in general we should avoid returning 500 to the client. 500 should mean the real error on the server.

Reply:
Yes, `return false` seems to work.

We will get a 500, if we don't edit the controller `#destroy` action.

I think our convention is to return 500, when destroy fails, perhaps to prevent attacker gaining information on what _will_ cause deletion?

Reply:
Actually, I'm still trying to get `return false` to work... currently it just always fails to delete (not conditional)